src/models/blockchain/block/mutator_set_update.rs

```diff
@@ -50,6 +57,15 @@ impl MutatorSetUpdate {
         &self,
         ms_accumulator: &mut MutatorSetAccumulator,
         removal_records: &mut [&mut RemovalRecord],
+    ) -> Result<()> {
+        self.apply_to_accumulator_and_records_inner(ms_accumulator, removal_records, true)
+    }
+
+    fn apply_to_accumulator_and_records_inner(
+        &self,
+        ms_accumulator: &mut MutatorSetAccumulator,
+        removal_records: &mut [&mut RemovalRecord],
+        check_for_double_spend: bool,
     ) -> Result<()> {
```

Comment:
This function interface strikes me as an unsafe potential pitfall.

Suggestion for improvement: The inner function should return a bool that indicates whether removing the items was successful. In the unsafe wrapper, you ignore this boolean. In the safe public wrapper you bail if it is true and return `Ok(())` otherwise.